### ISSUE TITLE
Add support for subphases from vt applications

### ISSUE THREAD

**Issue by PhilMiller** [DARMA-tasking/LB-analysis-framework#53]
See https://github.com/DARMA-tasking/vt/issues/708 for details of why this is desired, and the code that produces the stats files containing per-subphase timing data.

**Comment by lifflander** [comment]
Currently, the LB stats file outputs a format as so for each phase.

```
<iter/phase>, <object-id>, <time-in-seconds>
<iter/phase>, <object-id1-to/recv>, <object-id2-from/send>, <num-bytes>, <comm-type={1..6}>
```

In the new format, introduced by https://github.com/DARMA-tasking/vt/issues/708, we have added sub-phases to the format for the line with `<time-in-seconds>`. In particular, this commit https://github.com/DARMA-tasking/vt/commit/996d32ac465c0091e1acc208f2d2c2ad445afeff is where the format what changed. The new format looks like this:

```
<iter/phase>, <object-id>, <time-in-seconds> <#-of-subphases> '[' [<subphase-time-1>] ... [<subphase-time-N>] ']'
<iter/phase>, <object-id1-to/recv>, <object-id2-from/send>, <num-bytes>, <comm-type={1..6}>
```

We want to adjust the reader in the LBAF to read this format and process the new sub-phase data. Let's start with just getting it to read the data, then create the data structures in the simulator for processing the sub-phases.

- For changing the reader, look at this file: https://github.com/DARMA-tasking/LB-analysis-framework/blob/develop/src/IO/lbsLoadReaderVT.py
- You will also want to adjust the writer so it's able to write this back out: https://github.com/DARMA-tasking/LB-analysis-framework/blob/develop/src/IO/lbsLoadWriterVT.py

Let me know if you have questions. @nlslatt is running the app with the stats output so you have a valid input deck for the simulator to work with.

**Comment by bradybray** [comment]
@lifflander I cloned the repo and added a new branch for myself. I'll file a pull request once finished for review.

**Comment by lifflander** [comment]
@bradybray It turns out that we don't have the full application run with the new stats. That change didn't get merged onto the release branch that EMPIRE (the app) is using. Instead, I will attach some files of a benchmark locally so you have something to work with.

**Comment by lifflander** [comment]
Here's the link on building VT: https://darma-tasking.github.io/docs/html/vt-build.html

It doesn't mention it there, but if you don't want to rebuild the base image yourself, do a `docker-compose pull ubuntu-cpp-interactive` or `docker-compose pull ubuntu-cpp`

**Comment by lifflander** [comment]
By default, you get the settings in `vt/.env`:

```
REPO=lifflander1/vt
ARCH=amd64
UBUNTU=18.04
ULIMIT_CORE=0
COMPILER=gcc-7
COMPILER_TYPE=gnu
```

**Comment by lifflander** [comment]
Here's a set of stats file, that don't have any sub-phases for the `lb_iter` benchmark:
[stats.zip](https://github.com/DARMA-tasking/LB-analysis-framework/files/4841116/stats.zip)

**Comment by lifflander** [comment]
Phil says changing the VT printer format is totally fine. If you want to open a PR in VT to make it proper CSV, go for it

**Comment by bradybray** [comment]
> 
> 
> Here's a set of stats file that don't have any sub-phases for the `vt/examples/collection/lb_iter` benchmark:
> [stats.zip](https://github.com/DARMA-tasking/LB-analysis-framework/files/4841116/stats.zip)

@lifflander The subphase sections are empty. I sampled a random integer between 1 and 10 call it N. I then took N samples of a random normal and used this as simulated data. I doubt it truly represents the data intended to occupy these fields. But I assume it is ok for testing purposes. I was able to load the data and have now filled it with the randomly generated data. Are there any concerns about the random data? If not I'll continue.

**Comment by bradybray** [comment]
@PhilMiller @lifflander My latest commit adds support for reading the files with sub phase support given the sub phases contain data. Should the code also handle the instance where the suphases are empty or would this imply an error with whatever generates that data?

**Comment by marcinwrobel1986** [comment]
@lifflander @ppebay @PhilMiller What are we doing with this issue?

**Comment by ppebay** [comment]
I think this issue is now superseded/obsolete. @PhilMiller agreed?

**Comment by ppebay** [comment]
First step is to expand the reader to parse the subphase data and populate its output with it.

**Comment by ppebay** [comment]
Issue was superseded and resolved by #268 #269 #270
